crates/cadenza-gcode/STATUS.md

```diff
@@ -0,0 +1,67 @@
+# Status
+
+## Implementation Status
+
+### ✅ Completed
+
+- **Parser**: GCode lexer/parser producing Cadenza-compatible AST via Rowan CST
+- **Direct AST Construction**: No string generation or re-parsing
+- **Parameter Representation**: `[Letter, value]` structure (e.g., `X100` → `[X, 100]`)
+- **Comment Handling**: Comments preserved in CST as trivia
+- **Offset Tracking**: Accurate source positions for all tokens
+- **Snapshot Tests**: Auto-generated from test-data/*.gcode files
+- **Zero Allocations**: Iterator-based parsing without intermediate collections
+
+### 🎯 Architecture
+
+GCode is treated as an alternative syntax for Cadenza:
+- GCode commands → Apply nodes (function calls)
+- Parameters → Apply nodes with letter as receiver
+- Flags (no value) → Identifier nodes
+- Comments → Comment tokens in CST
+
+Example: `G1 X100 Y50` → `[G1, [X, 100], [Y, 50]]`
+
+Handler macros receive parameter expressions and can:
+- Pattern match on parameter names
+- Apply units based on command semantics
+- Handle optional parameters
+- Implement custom logic
+
+### 📋 Known Limitations
+
+1. **Basic GCode Only**: Currently parses simple command + parameter structure
+2. **No Checksums**: Doesn't validate or parse checksums (`*##` suffix)
+3. **Limited Error Recovery**: Basic error handling, could be more robust
+4. **No Macro Expansion**: GCode macros/variables not yet supported
+
+### 🚀 Future Enhancements
+
+1. **Extended GCode Support**:
+   - Checksums and validation
+   - Variable substitution
+   - Conditional execution
+   - Looping constructs
```

Comment:
Checksums seem good to have. The other stuff not so much. We're going to be getting gcode from slicers which are always straight line programs to my knowledge